### DISCUSSION TITLE
Support for EEPROM API function - EEPROM.get

### DISCUSSION THREAD

**Discussion by just-jason** [dbuezas/lgt8fx#196] [Ideas]
I would like to use a LGT8F328P in a project that uses the Arduino EEPROM.get function.   

https://www.arduino.cc/en/Reference/EEPROMGet   

Looking through the docs [here](https://github.com/dbuezas/lgt8fx/wiki/EEPROM-of-the-LGT8F328p-:-the-important-things-you-need-to-know), I do not see any mention of support for this function. Is there any chance to include support for this ?

**Comment by LaZsolt** [comment]
@just-jason 
Have you ever tried to use the `EEPROM.get()` function on an LGT8F328x board?
The definition of the `get()` is in the `EEPROM.h`, so it has to work. I think.

**Comment by LaZsolt** [comment]
@just-jason 

Plus, it will work when the next release will be finished. (v1.0.7)

**Reply by just-jason** [reply]
Is it possible that it only works in the 1.0.7 version?  I did try with current 1.0.6 and it does not work.

**Reply by LaZsolt** [reply]
Maybe you can do a test if you copy this two files
`https://github.com/dbuezas/lgt8fx/blob/master/lgt8f/libraries/E2PROM/EEPROM.cpp`
`https://github.com/dbuezas/lgt8fx/blob/master/lgt8f/libraries/E2PROM/EEPROM.h`
into this folder on your PC:
`C:\Users\->your username<-\AppData\Local\Arduino15\packages\LGT8fx Boards\hardware\avr\1.0.6\libraries\E2PROM`

**Reply by ithinkido** [reply]
Thanks for the tip. It is now compiling without issues. Do you know when this version will be released ?

**Reply by LaZsolt** [reply]
No, I don't know.

**Reply by vkvark** [reply]
I've tried a simple EEPROM.put and EEPROM.get sketch with the 2.01 core, and it didn't work - compilation ended with errors:

EEPROM_Test:8:8: error: 'class EEPROMClass' has no member named 'put'
EEPROM_Test:10:8: error: 'class EEPROMClass' has no member named 'get'

**Reply by SuperUserNameMan** [reply]
@vkvark : could you provide minimal compilable source code ?

**Reply by vkvark** [reply]
Sure, here you are:

```
#define USE_LGT_EEPROM_API
#include <EEPROM.h>
uint16_t addr = 0;
uint16_t val = 1270;

void setup() {
Serial.begin(57600);
EEPROM.put(addr, val);
delay(100);
EEPROM.get(addr, val);
Serial.println(val);
}

void loop() {
}
```

**Reply by SuperUserNameMan** [reply]
If you want to use the standard Arduino EEPROM API, you must not `#define USE_LGT_EEPROM_API`.

edit : `USE_LGT_EEPROM_API` is for backward compatibility with the original LGT EEPROM API (which was totally different from Arduino's )

**Reply by vkvark** [reply]
Thanks a lot, now it's working.
